src/libsystemd/sd-event/sd-event.c

```diff
@@ -1494,6 +1499,22 @@ _public_ int sd_event_source_set_io_fd(sd_event_source *s, int fd) {
         return 0;
 }
 
+_public_ int sd_event_source_get_io_fd_close(sd_event_source *s, bool *close) {
+        assert_return(s, -EINVAL);
+        assert_return(s->type == SOURCE_IO, -EDOM);
+
+        *close = s->io.close;
```

Comment:
hmm, i am a bit concerned about the name clash here, can you rename both the structure's field name and the argument here to something that is not obviously clashing with a C symbol name? old gcc even used to warn in this case, new gcc doesn't, but I think we should still avoid this already in order not to confuse ourselves...

Maybe call the field "close_on_free" or so, and the argument here just "b"?